### PR TITLE
feat: Implements tests and minor improvements

### PR DESCRIPTION
**Pull request by Xol**
Now there are a couple of tests for the whole parsing from Imgur. Also nsfw pictures are filtered out before it's saved to the database.

### CODE REVIEW

**Review by jbgerth — APPROVED** [review]
LGTM
Next we should talk about, how we can make the database integration a bit cleaner : )

File: src/data/memeDataAccess.ts [hunk 1]
```diff
@@ -8,7 +8,7 @@ const headers = {
   Accept: "application/json",
   Authorization: process.env.IMGUR_AUTHORIZATION ?? IMGUR.AUTHORIZATION,
 };
-const imageTypes: Set<string> = new Set(["image/jpeg", "image/png"]);
+const mimeImageTypes: Set<string> = new Set(["image/jpeg", "image/png"]);
```
Comment (coinzz) [inline src/data/memeDataAccess.ts:11]: Typo: Did you mean `memeImageTypes`?

Comment (coinzz) [reply src/data/memeDataAccess.ts:11]: Oh nevermind. I just didn't know that term.

**Review by coinzz — APPROVED** [review]
Looking good.

**Pull request merged**